astropy/io/ascii/tdat.py

```diff
@@ -515,19 +515,18 @@ def write(self, lines):
         lines.append("# Table Parameters")
         lines.append("#")
         for col in self.cols:
-            if str(col.info.dtype) in self._dtype_dict_out:
-                ctype = self._dtype_dict_out[str(col.info.dtype)]
-            elif "int" in str(col.info.dtype):
+            if str(col_type := col.info.dtype) in self._dtype_dict_out:
```

Comment:
Absolute nitpick, but with `col_type` used so much below, maybe just define it up front without the walrus?

Reply:
Might be neater, yes... pinging @d-giles and @esabol as original author and reviewer; I hope the remaining failed tests should be fixed now as well.

Reply:
Looks fine to me, but I'm only a casual Python programmer and unfamiliar with how to handle endian issues in Python.